src/query/tsdb/remote/client.go

```diff
@@ -197,7 +188,114 @@ func (c *grpcClient) FetchTags(
 	query *storage.FetchQuery,
 	options *storage.FetchOptions,
 ) (*storage.SearchResults, error) {
-	return nil, fmt.Errorf("remote fetch tags endpoint not supported")
+	pools, err := c.poolWrapper.WaitForIteratorPools(poolTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	request, err := encodeSearchRequest(query)
+	if err != nil {
+		return nil, err
+	}
+
+	// Send the id from the client to the remote server so that provides logging
+	id := logging.ReadContextID(ctx)
```

Comment:
tracing handles this automatically. Maybe add a todo to switch id propogation to opentracing?

Reply:
For rpc calls too? This is to send the request id over the wire so the server can track what happens to it

Reply:
yeah it does that for you